R/ihw_class.R

```diff
@@ -142,7 +162,12 @@ setMethod("weighted_pvalues", signature(object="ihwResult"),
 #---------------------------  covariate extraction ----------------------------------------------------------#
 
 covariates_ihwResult <- function(object){
-  object@df$covariate
+  covariate <- dplyr::select(object@df, dplyr::starts_with("covariate")) #TODO fix this
```

Comment:
`dplyr` is only listed as a suggested dependency. You cannot use it in regular code without checking that it is indeed available.

Reply:
You could replace the dplyr code by adapting the following snipped
``` r
dat <- data.frame(a_1 = 1:3, a_2 = 4:6, b_1 = letters[1:3], asdf = 0)
dat
#>   a_1 a_2 b_1 asdf
#> 1   1   4   a    0
#> 2   2   5   b    0
#> 3   3   6   c    0
sel_cols <- grepl("^a_", colnames(dat))
dat[,sel_cols]
#>   a_1 a_2
#> 1   1   4
#> 2   2   5
#> 3   3   6
```